File: Python/7py 2/reverse_esrever.py
"""
|--------------------------------------------------------------------------
| esreveR
|--------------------------------------------------------------------------
|
"""

def reverse(lst):
   res = list()
   for i in range(len(lst)): 
      res.append(lst.pop())
   return res


def reverse(lst):
   res = list()
   for x in lst:
      res.insert(0, x)
   return res


def reverse(lst):
   res = list()
   for i in range(len(lst)-1,-1,-1):
      res.append(lst[i])
   return res


print(reverse(list([1,2,3])), [3,2,1])
print(reverse(list([1,None,14,"two"])), ["two",14,None,1])

File: JavaScript/7js 3/reverse_esrever.js
/*
|--------------------------------------------------------------------------
| esreveR
|--------------------------------------------------------------------------
|
*/

function reverse(array, res = []) {
   for (let i = array.length - 1; i >= 0; i--) res.push(array[i]);
   return res;
}


function reverse(array) {
   return array.reduce((pre, val) => [val, ...pre], []);
}


function reverse(array) {
   return array.map((c, i) => array[array.length - 1 - i]);
}


function reverse(array) {
   return array.reduce(function (a, b) {
      return [b].concat(a);
   }, []);
}


console.log(reverse([1, 2, 3]), [3, 2, 1]);
console.log(reverse([1, null, 14, "two"]), ["two", 14, null, 1]);